web-ui/src/app/services/results.service.ts

```diff
@@ -152,30 +167,29 @@ export class ResultsService {
         return false;
     }
 
-    async highlightSentenceTree(sentenceId: string, treebank: string, nodeIds: number[], database: string = null) {
-        const url = await this.configurationService.getApiUrl(
-            'tree', [
-                treebank,
-                sentenceId,
-                nodeIds.join('-')],
-            { ...(database && { db: database }) });
+    highlightSentenceTree(sentenceXml: string, nodeIds: number[]) {
```

Comment:
Dit is helaas niet volledig: de hit kan een subset zijn van de gehele zin. Terwijl het hier juist wenselijk is om de hele zin te laten zien en wat daarin matcht. Of geeft de server nu wel de structuur voor de hele zin terug voor iedere match?

Reply:
Bedoel je dat de sentenceXml niet de volledige tree is? In de server heb ik daar verder geen veranderingen in gemaakt, maar ik heb geen duidelijk incomplete zinnen voorbij zien komen (al is dat misschien toeval dan).

Reply:
Een duidelijk voorbeeld is deze: http://gretel.hum.uu.nl/gretel4/ng/xpath-search?currentStep=2&xpath=%2F%2Fnode%5B@id%3D%2215%22%20and%20%0A%20%20%20%20node%5B@word%3D%22over%22%5D%20and%20%0A%20%20%20%20node%5Bnode%5B@word%3D%22de%22%5D%20and%0A%20%20%20%20%20%20%20%20not(node%5B@word%3D%22hele%22%5D)%20and%0A%20%20%20%20%20%20%20%20node%5B@word%3D%22wereld%22%5D%5D%5D&selectedTreebanks=%7B%22corpus%22:%22lassy%22,%22components%22:%5B%22DPC%22,%22WIKI%22,%22WRPE%22,%22WRPP%22,%22WSU%22%5D%7D&retrieveContext=0

Hier wordt alleen een kleine XML teruggeven: 
```xml
<node begin="6" cat="pp" end="9" id="15" rel="mod">
    <node begin="6" end="7" id="16" lemma="over" pos="prep" postag="VZ(init)" pt="vz" rel="hd" root="over" vztype="init" word="over"/>
    <node begin="7" cat="np" end="9" id="17" rel="obj1">
      <node begin="7" end="8" id="18" lemma="de" lwtype="bep" naamval="stan" npagr="rest" pos="det" postag="LID(bep,stan,rest)" pt="lid" rel="det" root="de" word="de"/>
      <node begin="8" end="9" genus="zijd" getal="ev" graad="basis" id="19" lemma="wereld" naamval="stan" ntype="soort" pos="noun" postag="N(soort,ev,basis,zijd,stan)" pt="n" rel="hd" root="wereld" word="wereld"/>
    </node>
  </node>
```
terwijl de gehele XML veel groter is.

Reply:
Loopt weer via de server :)